deployment/terraform-django/app.tf

```diff
@@ -141,9 +141,11 @@ resource "aws_ecs_service" "app" {
   enable_execute_command = true
   force_new_deployment   = true
 
+
   network_configuration {
-    security_groups = [aws_security_group.app.id]
-    subnets         = aws_subnet.private.*.id
+    security_groups  = [aws_security_group.app.id]
+    subnets          = aws_subnet.public.*.id
+    assign_public_ip = true
```

Comment:
A further thought: it looks like we might be able to achieve the same goal while keeping the application instances in the private subnet by expanding our use of VPC endpoints. https://docs.aws.amazon.com/vpc/latest/privatelink/create-interface-endpoint.html . My thinking is that while this application in particular doesn't store sensitive information, some applications we work on do and those clients may not be comfortable with the use of public subnets, so if we're looking for a general, NAT-less solution we may want to stick with private subnets (even though I think the current solution is fine for this particular project; the security groups keep them from being wide open to the world).

The main downside to a VPC endpoint that I can see is that they're not free. However, they're a lot cheaper than NAT Gateways because they cost $0.01/hr + $0.01/GB compared to $0.045/hr + $0.045/GB. The other downside is that they're more complicated -- we'll have to pick and choose based on the services used by each project if we don't want to get charged unnecessarily.

Reply:
It did have Public IP previously - two things that break would be rollbar and ECR access (although that is likely fixable with VPC Endpoints)

It was not obvious to me that you could actually make VPC endpoints for rollbar or non-AWS services.

Reply:
Oops, no, I just forgot about Rollbar 🤦 So this may be a situation where there's not a good one-size-fits-all solution, because yeah I don't see a way to connect to Rollbar from inside a VPC without a NAT Gateway.